mkdocs.yml

```diff
@@ -25,6 +25,22 @@ theme:
   features:
     - content.code.copy
     - navigation.indexes
+  palette:
+    - media: "(prefers-color-scheme)"
+      toggle:
+        icon: material/brightness-auto
+        name: Switch to light mode
+    - media: "(prefers-color-scheme: light)"
+      scheme: default 
+      toggle:
+        icon: material/brightness-7
+        name: Switch to dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      toggle:
+        icon: material/brightness-4
+        name: Switch to system preference
```

Comment:
<img width="986" alt="image" src="https://github.com/user-attachments/assets/5c091ed4-ed21-4125-aa77-18ff123b30de">
how to set auto mode working? I always switch between two light themes and one dark theme

Reply:
> Newer operating systems allow to automatically switch between light and dark appearance during day and night times

this is mention in the documentation of mkdocs.

Reply:
It does not seem to be working to me. How have you tested this?

Reply:
For me changing it into 
```
  palette:
    # Palette toggle for light mode
    - media: "(prefers-color-scheme: light)"
      scheme: default
      toggle:
        icon: material/brightness-7
        name: Switch to dark mode

    # Palette toggle for dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/brightness-4
        name: Switch to light mode
```
worked, so unless you change theme you got the one from media query, you can test via that dev tool but remember to delete from application > local storage > /addonfactory-ucc-generator/.__palette

Reply:
I am able to change from dark to light and vice-versa, but as same as viktor, I can only have the light theme when auto mode is selected.

I have gone through many github issues for the same, here are the references: 
- https://github.com/squidfunk/mkdocs-material/issues/3123
- https://github.com/squidfunk/mkdocs-material/issues/3244
- https://github.com/squidfunk/mkdocs-material/discussions/3911

Reply:
Also I have checked that, the object saved in local storage of official squidfunk docs is:

```
{
    "media": "(prefers-color-scheme: dark)",
    "scheme": "slate",
    "primary": "black",
    "accent": "indigo"
}
```

but in our local storage, object is:

```
{
    "scheme": "default",
    "primary": "indigo",
    "accent": "indigo"
}
```
I am debugging why there is difference in the object.

Reply:
what i ment 

> For me changing it into
> 
> ```
>   palette:
>     # Palette toggle for light mode
>     - media: "(prefers-color-scheme: light)"
>       scheme: default
>       toggle:
>         icon: material/brightness-7
>         name: Switch to dark mode
> 
>     # Palette toggle for dark mode
>     - media: "(prefers-color-scheme: dark)"
>       scheme: slate
>       toggle:
>         icon: material/brightness-4
>         name: Switch to light mode
> ```
> 
> worked, so unless you change theme you got the one from media query, you can test via that dev tool but remember to delete from application > local storage > /addonfactory-ucc-generator/.__palette

yea, with that change there isnt automatic mode, but as sson as you load documents the theme is adjusted to current settings, so if you use dark mode it will use it and if you have light it will use it also on the very first load.

then it is saved in local storage as a variable, not the best solution as it would be good to have it update asap

Reply:
We use mkdocs-material@9.2.7
https://github.com/splunk/addonfactory-ucc-generator/blob/76dd82f4f158cb1dcb32eaab845c10cd14d62fce/poetry.lock#L856

For this version, automatic lights / dark mode was for Sponsors only
https://github.com/squidfunk/mkdocs-material/blob/9.2.7/docs/setup/changing-the-colors.md#automatic-light--dark-mode

We cannot update further since it requires newer version of python, hence we cannot use automatic light / dark mode out of the box

Reply:
removed the automatic mode, and kept only light and dark.